### PR TITLE
MNT mention #273 improvements in the changelog

### PR DESCRIPTION
**Pull request by pierreglaser**
Pickling classes - try for example with this script:
```python
import cloudpickle


class A:
    pass


cloudpickle.dumps(A)
```

is now slightly faster (~10%) thanks to #273, so it is worth mentioning it in the changelog.

### CODE REVIEW

**Comment by pierreglaser** [comment]
(Rebased)

File: CHANGES.md [hunk 1]
```diff
@@ -9,6 +9,10 @@
   but is no longer officially supported by cloudpickle)
   ([issue #276](https://github.com/cloudpipe/cloudpickle/pull/276))
 
+- Internal refactoring of the pickling logic to make the code more predictable.
```
Comment (ogrisel) [inline CHANGES.md]: Maybe be more specific an some how mention that this is about the detection of dynamic classes?

File: CHANGES.md [hunk 2]
```diff
@@ -9,6 +9,10 @@
...
+  This refactoring also yielded small performance improvements (~10%)
```
Comment (ogrisel) [inline CHANGES.md]: yielded => yield

**Comment by ogrisel** [comment]
Thanks, merged.

**Pull request merged**